Packs/GigamonThreatINSIGHT/Integrations/GigamonThreatINSIGHT/GigamonThreatINSIGHT.py

```diff
@@ -0,0 +1,495 @@
+                rule_category = rule['category']
+                rule_signature = rule['query_signature']
+        detection.update({'rule_name': rule_name})
+        detection.update({'rule_description': rule_description})
+        detection.update({'rule_severity': rule_severity})
+        detection.update({'rule_confidence': rule_confidence})
+        detection.update({'rule_category': rule_category})
+        detection.update({'rule_signature': rule_signature})
+    return r_json
+
+
+def responseToEntry(r_json, path, title):
+    # set data and type
+    data = []
+    data_type = title.lower().replace(" ", "_")
+    # set context path
+    contextPath = "Insight." + path
+    context = {contextPath: []}  # type: dict
+    # check for exclusions
+    if data_type == 'device_list':
+        r_json = r_json['devices']
+    # iterate based on data type
+    if data_type in ('data'):
+        for i in range(0, len(r_json[data_type])):
+            new_item = {}
+            if isinstance(r_json[data_type][i], list):
+                for j in range(0, len(r_json[data_type][i])):
+                    new_pair = {r_json['columns'][j]: r_json[data_type][i][j]}
+                    new_item.update(new_pair)
+            elif isinstance(r_json[data_type][i], dict):
+                for item in r_json[data_type][i]:
+                    if isinstance(r_json[data_type][i][item], int):
+                        new_pair = {str(item): int(r_json[data_type][i][item])}
+                    else:
+                        new_pair = {str(item): str(r_json[data_type][i][item])}
+                    new_item.update(new_pair)
+            data.append(new_item)
+            context[contextPath].append(createContext(new_item))
+    elif data_type in ('pcap_task', 'summary', 'file'):
+        new_item = {}
+        for item in r_json[data_type]:
+            if isinstance(r_json[data_type][item], dict):
+                r_json[data_type][item] = flattenDict(r_json[data_type][item])
+            if isinstance(r_json[data_type][item], list):
+                r_json[data_type][item] = flattenList(r_json[data_type][item])
+            new_pair = {item: str(r_json[data_type][item])}
+            new_item.update(new_pair)
+        data.append(new_item)
+        context[contextPath].append(createContext(new_item))
+    else:
+        for item in r_json[data_type]:
+            new_item = {}
+            for field in item:
+                # if there are any unicode characters, encode them to utf8
+                if isinstance(item[field], str):
+                    item[field] = str(item[field].encode("utf-8"))
+                new_pair = {field: str(item[field])}
+                new_item.update(new_pair)
+            data.append(new_item)
+            context[contextPath].append(createContext(new_item))
+    # submit results
+    demisto.results({'ContentsFormat': formats['table'], 'Type': entryTypes['note'], 'Contents': data, 'EntryContext': context})
```

Comment:
There is no need for this function, you have all the utility functions that are needed in the [CommonServerPython](https://xsoar.pan.dev/docs/reference/api/common-server-python).
```suggestion
```